### PR TITLE
Add Oauth2 Support And  Add Path Support in sdk 

### PR DESCRIPTION
**Pull request by zxy-lgtm**
there also have some problems :
I use code flow to get a oauth2 access_token so that we can refresh it to enable long-term use of the API in some scenarios.
But in the first step, I can only access get on the ` page_ code_ url `method to obtain the corresponding'`code`in the redirection. The` code `is attached to the redirection URL set by Azure. I think you can obtain the `code` by obtaining the redirected URL, but it is not successful.
https://docs.microsoft.com/en-us/onedrive/developer/rest-api/getting-started/msa-oauth?view=odsp-graph-online

### CODE REVIEW

**Comment by xxchan** [comment]
Hi, about oauth2, we can let the user decide how to generate the token and simply pass it as the [comment](https://github.com/beyondstorage/go-service-gdrive/issues/17#issuecomment-895728399) says and the [PR](https://github.com/beyondstorage/go-service-gdrive/pull/18) does. So `utils_token` is not needed (but you can use it to test for yourself).

(`utils_drives.go` not yet reviewed

BTW, A suggestions: Use descriptive PR title (what problem you solves). In the PR description, you can say `closes #11` (or `ref #11` if the PR only resolves part of the issue).

File: utils_token.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	get_code_url  = "https://login.live.com/oauth20_authorize.srf?response_type=code&redirect_uri=http://localhost:9999&response_mode=query&scope=offline_access%20user.read%20mail.read%20Files.ReadWrite.All&client_id=8ce1780f-12cc-45eb-a4c9-83f284e56621"
+	get_token_url = "https://login.live.com/oauth20_token.srf"
+)
+
+//for test we can use the client I have registered
+const (
+	secret       = "-uGEL0oQZ86Wwr2I-dTF3~Y3i6sCGE5z~n"
+	client_id    = "8ce1780f-12cc-45eb-a4c9-83f284e56621"
+	redirect_url = "http://localhost:9999"
+)
```
Comment (xxchan) [inline utils_token.go]: Generally these sensitive data should not be pushed to GitHub. (You can use GitHub secrets when necessary.) And it seems that we don't need `utils_token` as https://github.com/beyondstorage/go-service-onedrive/pull/12#issuecomment-898895801 says

Comment (xxchan) [reply utils_token.go]: You can try to remove this file from commit history... (a chance to learn advanced git operation 🤣

**Comment by junaire** [comment]
Hello, @zxy-lgtm, looks like you are in the trouble. Maybe I can give you some suggestions:
- We prefer create new branches every time open a PR, see [beyondstorage/go-storage/pull/697#issuecomment](https://github.com/beyondstorage/go-storage/pull/697#issuecomment-888200028)
- Read code about `go-storage` and `go-service-*`. I highly recommend you to read [go-service-gdrive](https://github.com/beyondstorage/go-service-gdrive) and [go-service-s3](https://github.com/beyondstorage/go-service-s3). You can find that many codes are very similar or even duplicated. So implement `go-service-*` is not particularly complicated.
- Read code about onedrive. I just did a simple search, you can read [koofr/go-onedriveclient](https://github.com/koofr/go-onedriveclient), or [rclone](https://github.com/rclone/rclone), they are really good examples.
- Don't hesitate to ask for help if you are in trouble. You can reach us on github issues, telegram, or matrix. Everyone in the commnity is happy to help you.

At last, happy hacking! ;-)

**Comment by xxchan** [comment]
What about pushing `Storager` implementations together, instead of only the utils? (i.e., how do you use these utils)

**Comment by Xuanwo** [comment]
Hello, any updates here?

**Comment by xxchan** [comment]
Hello, any updates? (Can you tell us your schedule?)

**Comment by zxy-lgtm** [comment]
I will try to finish it before 9.10



---Original---
From: ***@***.***&gt;
Date: Mon, Sep 6, 2021 14:14 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [beyondstorage/go-service-onedrive] Add Oauth2 Support And  Add Path Support in sdk  (#12)




 
Hello, any updates? (Can you tell us your schedule?)
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Pull request closed without merging**